api/types/saml.go

```diff
@@ -134,14 +134,20 @@ func (o *SAMLConnectorV2) SetResourceID(id int64) {
 
 // WithoutSecrets returns an instance of resource without secrets.
 func (o *SAMLConnectorV2) WithoutSecrets() Resource {
-	k := o.GetSigningKeyPair()
-	if k == nil {
-		return o
+	k1 := o.GetSigningKeyPair()
+	k2 := o.GetEncryptionKeyPair()
+	var q1, q2 *AsymmetricKeyPair
+	if k1 != nil {
+		*q1 = *k1
```

Comment:
Doesn't this still attempt to write to the `q1 = nil` address...?

I think you should do something like
```go
var q1, q2 AsymmetricKeyPair
// ...
q1 = *k1
// ...
o2.SetSigningKeyPair(&q1)
```

Is this function `WithoutSecrets` run as part of any tests?

Reply:
Hmm, no I don't think we have universal component tests. I'll add some for this one at least.